src/synapse/workload.py

```diff
@@ -545,6 +571,117 @@ def create_mjolnir_config(container: ops.Container, access_token: str, room_id:
 
 
+def _get_irc_bridge_config(server_name: str, db_connect_string: str) -> typing.Dict:
+    """Create config as expected by irc bridge.
+
+    Args:
+        server_name: server name of the Synapse instance.
+        db_connect_string: database connection string.
+
+    Returns:
+        IRC Bridge configuration
+    """
+    with open("templates/irc_bridge_production.yaml", encoding="utf-8") as irc_config_file:
+        config = yaml.safe_load(irc_config_file)
+        config["homeserver"]["url"] = SYNAPSE_URL
+        config["homeserver"]["domain"] = server_name
+        config["database"]["connectionString"] = db_connect_string
+        return config
+
+
+def create_irc_bridge_config(
+    container: ops.Container, server_name: str, db_connect_string: str
+) -> None:
+    """Create irc bridge configuration.
+
+    Args:
+        container: Container of the charm.
+        server_name: server name of the Synapse instance.
+        db_connect_string: database connection string.
+
+    Raises:
+        CreateIRCBridgeConfigError: something went wrong creating irc bridge config.
+    """
+    try:
+        config = _get_irc_bridge_config(
+            server_name=server_name, db_connect_string=db_connect_string
+        )
+        container.push(IRC_BRIDGE_CONFIG_PATH, yaml.safe_dump(config), make_dirs=True)
+    except ops.pebble.PathError as exc:
+        raise CreateIRCBridgeConfigError(str(exc)) from exc
+
+
+def _get_irc_bridge_app_registration(container: ops.Container) -> None:
+    """Create registration file as expected by irc bridge.
+
+    Args:
+        container: Container of the charm.
+
+    Raises:
+        WorkloadError: something went wrong creating irc bridge registration.
+    """
+    registration_result = _exec(
+        container,
+        [
+            "/bin/node",
+            "/app/app.js",
+            "-r",
+            "-f",
+            IRC_BRIDGE_REGISTRATION_PATH,
+            "-u",
+            "http://localhost:",
+            str(SYNAPSE_PORT),
+            "-c",
+            IRC_BRIDGE_CONFIG_PATH,
+            "-l",
+            "my_bot",
+        ],
+    )
+    if registration_result.exit_code:
+        logger.error(
+            "creating irc app registration failed, stdout: %s, stderr: %s",
+            registration_result.stdout,
+            registration_result.stderr,
+        )
+        raise WorkloadError("Creating irc app registration failed, please check the logs")
+
+
+def create_irc_bridge_app_registration(container: ops.Container) -> None:
+    """Create irc bridge app registration.
+
+    Args:
+        container: Container of the charm.
+
+    Raises:
+        CreateIRCBridgeRegistrationError: error creating irc bridge app registration.
+    """
+    try:
+        _get_irc_bridge_app_registration(container=container)
+        _add_app_service_config_field(container=container)
+    except ops.pebble.PathError as exc:
+        raise CreateIRCBridgeRegistrationError(str(exc)) from exc
```

Comment:
It is kind of confusing that calling a non-ops function might raise a `ops` error. I'm guessing these bubble up, perhaps handle them directly in the function?